src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1079,6 +1080,10 @@ private void SetProjectCurrentDirectory()
 
             _projectLoggingContext = _nodeLoggingContext.LogProjectStarted(_requestEntry);
 
+            // Now that the project has started, parse a few known properties which indicate warning codes to treat as errors or messages
+            //
+            ConfigureWarningsAsErrorsAndMessages();
```

Comment:
Hopefully this is the only place where MSBuild evaluates a project file during build. :)